src/Autosuggest.js

```diff
@@ -274,10 +280,19 @@ class Autosuggest extends Component {
       }
     };
     const onMouseEnter = (event, { sectionIndex, itemIndex }) => {
-      updateFocusedSuggestion(sectionIndex, itemIndex);
+      if (!focusedUsingKeyboard) {
+        updateFocusedSuggestion(sectionIndex, itemIndex);
+      }
     };
     const onMouseLeave = () => {
-      updateFocusedSuggestion(null, null);
+      if (!focusedUsingKeyboard) {
+        updateFocusedSuggestion(null, null);
+      }
+    };
+    const onMouseMove = () => {
+      if (focusedUsingKeyboard) {
+        updateFocusedUsingKeyboard(false);
+      }
```

Comment:
I wonder if we can avoid having the `onMouseMove` handler. This handler would potentially be called hundreds of times if user "reviews" the suggestions by moving the mouse downwards. By introducing `onMouseMove` we penalise all users that don't have a scrollbar.